examples/newsreaderExample/src/main/java/io/realm/examples/newsreader/model/Repository.java

```diff
@@ -0,0 +1,140 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.examples.newsreader.model;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.UiThread;
+
+import java.io.Closeable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import io.realm.Realm;
+import io.realm.RealmResults;
+import io.realm.Sort;
+import io.realm.examples.newsreader.NewsReaderApplication;
+import io.realm.examples.newsreader.R;
+import io.realm.examples.newsreader.model.entity.NYTimesStory;
+import io.realm.examples.newsreader.model.network.NYTimesDataLoader;
+import rx.Observable;
+import rx.functions.Func1;
+import rx.subjects.BehaviorSubject;
+import timber.log.Timber;
+
+/**
+ * Class for handling loading and saving data.
+ *
+ * A repository is a potentially expensive resource to have in memory, so should be closed when no longer needed/used.
+ *
+ * @see <a href="http://martinfowler.com/eaaCatalog/repository.html">Repository pattern</a>
+ */
+public class Repository implements Closeable {
+
+    private static final long MINIMUM_NETWORK_WAIT_SEC = 120; // Minimum 2 minutes between each network request
+
+    private final Realm realm;
+    private final NYTimesDataLoader dataLoader;
+    private final String apiKey;
+    private Map<String, Long> lastNetworkRequest = new HashMap<>();
+    private BehaviorSubject<Boolean> networkLoading = BehaviorSubject.create(false);
+
+    @UiThread
+    public Repository() {
+        realm = Realm.getDefaultInstance();
+        dataLoader = new NYTimesDataLoader();
+        apiKey = NewsReaderApplication.getContext().getString(R.string.nyc_top_stories_api_key);
+    }
+
+    /**
+     * Keeps track of the current network state.
+     *
+     * @return {@code true} if the network is currently being used, {@code false} otherwise.
+     */
+    public Observable<Boolean> networkInUse() {
+        return networkLoading.asObservable();
+    }
+
+    /**
+     * Loads the news feed as well as all future updates.
+     */
+    public Observable<RealmResults<NYTimesStory>> loadNewsFeed(@NonNull String sectionKey, boolean forceReload) {
+        // Start loading data from the network if needed
+        // It will put all data into Realm
+        if (forceReload || timeSinceLastNetworkRequest(sectionKey) > MINIMUM_NETWORK_WAIT_SEC) {
+            dataLoader.loadData(sectionKey, apiKey, realm, networkLoading);
+            lastNetworkRequest.put(sectionKey, System.currentTimeMillis());
+        }
+
+        // Return the data in Realm. The query result will be automatically updated when the network requests
+        // save data in Realm
```

Comment:
Minor issue, maybe create a separate method to do reload? To avoid creating overhead RealmResults.